packages/flutter/lib/src/gestures/resampler.dart

```diff
@@ -0,0 +1,424 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:collection';
+
+import 'events.dart';
+
+/// Class for pointer event resampling.
+///
+/// An instance of this class can be used to resample one sequence
+/// of pointer events. Multiple instances are expected to be used for
+/// multi-touch support. The sampling frequency and the sampling
+/// offset is determined by the caller.
+///
+/// This can be used get smooth touch event processing on devices
+/// without high frequency touch sensors. Resampling adds event
+/// processing latency and should be avoided when low-latency is
+/// critical.
+///
+/// Resampling is currently limited to event position.
```

Comment:
Nit: position and delta?

Reply:
And is there anything else that may require resampling? One thing I can think of is `scrollDelta` for `PointerScrollEvent`, which looks like quite different logic than what's here. It might worth filing an issue to record that.

Reply:
Done